extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -20,126 +19,134 @@ namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ;
 
 internal sealed class RabbitMQListener : IListener, IScaleMonitor<RabbitMQTriggerMetrics>
 {
+    private const int ListenerNotStarted = 0;
+    private const int ListenerStarting = 1;
+    private const int ListenerStarted = 2;
+    private const int ListenerStopping = 3;
+    private const int ListenerStopped = 4;
```

Comment:
Maybe replace with enum?

Reply:
It's not possible to use `Interlocked.CompareExchange` on enum values without workarounds. There is a GitHub issue on this: https://github.com/dotnet/runtime/issues/65184.

Reply:
I somewhat figured when I saw that later in the file. Looks like the cleanest way to do it without some inane workaround around `Interlocked.CompareExchange`, which is not really worth it.